packages/react-native/React/Views/RCTView.m

```diff
@@ -785,6 +786,10 @@ - (void)reactSetFrame:(CGRect)frame
   [super reactSetFrame:frame];
   if (!CGSizeEqualToSize(self.bounds.size, oldSize)) {
     [self.layer setNeedsDisplay];
+    // Update transform for transform origin due to change in view dimension
+    if (self.transformOrigin.length > 0) {
+      self.layer.transform = [RCTConvert CATransform3D:self.rawTransform viewWidth:self.bounds.size.width viewHeight:self.bounds.size.height transformOrigin: self.transformOrigin];
```

Comment:
I think you can use [anchorPoint](https://developer.apple.com/documentation/quartzcore/calayer/1410817-anchorpoint) instead. Note that it's a percent value rather than pixel, so if you do this you'll need to flip your logic so your non-percent values are **multiplied** by the view width/height

Reply:
initial version i tried that used anchorPoint but anchorPoint also [changes the position](https://stackoverflow.com/questions/1968017/changing-my-calayers-anchorpoint-moves-the-view) of the view and it might conflict with yoga's setting position or will need some handling so I dropped that idea, also Android has pivot x and pivot y but no pivot z so thought of using the raw translates to achieve transform-origin and keeping it consistent on android and iOS. 😅